File: backend-server/src/main/java/cz/cuni/mff/kocaro/comm_app/comm_app_backend/dto/NvcScenarioResponseDto.java
package cz.cuni.mff.kocaro.comm_app.comm_app_backend.dto;

import java.util.List;

/**
 * The JSON payload sent to the client when it requests a scenario.
 */
public record NvcScenarioResponseDto(
        Long id,
        String title,
        String contextDescription,
        List<NvcScenarioOptionDto> options
) {}

File: CommAppAndroid/app/src/main/java/cz/cuni/mff/kocaro/comm_app/commappandroid/network/dto/NvcScenarioResponseDto.kt
package cz.cuni.mff.kocaro.comm_app.commappandroid.network.dto

import com.google.gson.annotations.SerializedName

/**
 * DTO for NVC Scenario exercises
 */
data class NvcScenarioResponseDto(
    @SerializedName("id") val id: Long,
    @SerializedName("title") val title: String,
    @SerializedName("contextDescription") val contextDescription: String,
    @SerializedName("options") val options: List<NvcScenarioOptionDto>
)
